solvebio/resource/datasetcommit.py

```diff
@@ -20,11 +21,17 @@ class DatasetCommit(CreateableAPIResource, ListableAPIResource,
         ('created_at', 'Created'),
     )
 
+    @property
     def dataset(self):
-        self._client.Dataset.retrieve(self['dataset'])
+        response = self._client.get(self['dataset']['url'], {})
+        return convert_to_solve_object(response, client=self._client)
```

Comment:
Do you think it might be sufficient to do:

`return convert_to_solve_object(self['dataset'], client=self._client)`?